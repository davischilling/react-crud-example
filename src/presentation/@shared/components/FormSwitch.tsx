import { FormControlLabel, FormGroup, Grid, Switch, SwitchProps } from '@mui/material';

type FormSwitchProps = {
  label: string;
} & SwitchProps;

export default function FormSwitch({ label, ...rest }: FormSwitchProps) {
  return (
    <Grid item xs={12}>
      <FormGroup>
        <FormControlLabel control={<Switch {...rest} />} label={label} />
      </FormGroup>
    </Grid>
  );
}
