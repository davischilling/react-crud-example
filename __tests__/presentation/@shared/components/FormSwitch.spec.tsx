import { render } from '@testing-library/react';
import FormSwitch from '../../../../src/presentation/@shared/components/FormSwitch';

describe('FormSwitch', () => {
  it('should render the FormSwitch', () => {
    const { asFragment } = render(
      <FormSwitch
        label="active"
        name="is_active"
        color="secondary"
        onChange={() => {}}
        checked={true}
        inputProps={{ 'aria-label': 'controlled' }}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
