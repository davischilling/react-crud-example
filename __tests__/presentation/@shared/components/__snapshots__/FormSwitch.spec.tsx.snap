// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSwitch should render the FormSwitch 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked Mui-checked css-priqlz-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              aria-label="controlled"
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="is_active"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          active
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
